service/src/main/java/io/camunda/service/UserServices.java

```diff
@@ -39,8 +39,8 @@ public SearchQueryResult<UserEntity> search(final UserQuery query) {
   }
 
   @Override
-  public UserServices<T> withAuthentication(final Authentication authentication) {
-    return new UserServices<>(brokerClient, searchClient, transformers, authentication);
+  public UserServices withAuthentication(final Authentication authentication) {
+    return new UserServices(brokerClient, searchClient, transformers, authentication);
   }
 
   public CompletableFuture<UserRecord> createUser(
```

Comment:
❓As now we have a record for CreateUserRequest, isn't it better to change the signature of this method and pass the  CreateUserRequest as the arg?